### PR TITLE
Update px cli to make `--cloud_addr` a required argument

### PR DESCRIPTION
**Pull request by ddelnano**
Summary: Update px cli to make `--cloud_addr` a required argument

This removes the CC4P dependency on the Pixie cli so that the project is agnostic to any vendor provided Pixie cloud. While this argument is now required, if the cli is run within an interactive shell, the user will be presented with the available Pixie clouds and have the opportunity to select the one they want to use (see Test Plan below).

My plan is to take this a step further for the interactive case and have the cli populate the `~/.pixie/config.json` (or similar) file if the user wants to save their selected cloud. Future invocations of the cli will use what is stored in `~/.pixie`.

Relevant Issues: N/A

Type of change: /kind cleanup

Test Plan: Tested the following scenarios
- [x] `px help`, `px --help`, `px deploy --help` (command) and `px script help` (command with sub commands) bypass the new required logic. Note: `px deploy help` would require the argument since it is an invalid command (`deploy` doesn't have a help subcommand)
- [x] Running non interactively exits if `--cloud_addr` has not been provided
```
$ ssh host-with-pixie-cli "/bin/bash -c '~/code/pixie/px deploy help'"
Pixie CLI
No cloud address provided. Please set the cloud address using the `--cloud_addr` flag or `PX_CLOUD_ADDR` environment variable.
```

- [x] Running interactively shows the list of possible clouds
```
$ bazel run src/pixie_cli:px -- deploy
INFO: Analyzed target //src/pixie_cli:px (0 packages loaded, 0 targets configured).
INFO: Found 1 target...
Target //src/pixie_cli:px up-to-date:
  bazel-bin/src/pixie_cli/px_/px
INFO: Elapsed time: 0.309s, Critical Path: 0.02s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action
INFO: Running command line: bazel-bin/src/pixie_cli/px_/px deploy
Pixie CLI
Use the arrow keys to navigate: ↓ ↑ → ←
? Select Pixie cloud:
  ▸ withpixie.ai:443
```
- [x] Running command that requires cloud addr succeeds

Changelog Message: Update `px` cli to make `--cloud_addr` a required argument if not populated by the `PX_CLOUD_ADDR` environment variable

### CODE REVIEW

File: src/pixie_cli/pkg/cmd/run.go [hunk 1]
```diff
@@ -39,13 +39,14 @@ import (
 )
 
 func init() {
+	cloudAddr := viper.GetString("cloud_addr")
```
[the remark this answers is not in the document]
Comment (ddelnano) [reply src/pixie_cli/pkg/cmd/run.go:42]: No, I didn't consider if the root flag was bound and read before this init function. Removing it from the help text sounds like a much better option.

**Comment by ddelnano** [comment]
@vihangm this is ready for another round of review when you have the chance.

File: src/pixie_cli/pkg/cmd/root.go [hunk 2]
```diff
@@ -185,6 +198,42 @@ var RootCmd = &cobra.Command{
...
+	// Commands within allow list should be opted out in addition to Cobra's
+	// default help command
+	if slices.Contains(cmdsCloudAddrNotReqd, cmd) || cmd.Short == "Help about any command" {
+		return defaultCloudAddr
+	}
+
+	cloudAddr := viper.GetString("cloud_addr")
+	if cloudAddr == "" {
+		if !isatty.IsTerminal(os.Stdin.Fd()) {
+			utils.Errorf("No cloud address provided during run within interactive shell. Please set the cloud address using the `--cloud_addr` flag or `PX_CLOUD_ADDR` environment variable.")
```
Comment (vihangm) [inline src/pixie_cli/pkg/cmd/root.go:217]: error says `interactive shell` but this isn't a tty right?

Comment (ddelnano) [reply src/pixie_cli/pkg/cmd/root.go]: Good call, I meant to say non-interactive shell.

**Review by vihangm — APPROVED** [review]

**Pull request merged**